src/draw.py

```diff
@@ -0,0 +1,166 @@
+        self.nodes_data = self.add_coords(coords)
+        self.edges_data = self.graph.get_edges()
+        
+        self.labels_color = labels_color
+        self.undiscovered_node_color=undiscovered_node_color 
+        self.discovered_node_outline=discovered_node_outline
+        self.current_node_fill=current_node_fill
+        self.discovered_node_fill=discovered_node_fill
+        self.undiscovered_node_fill=undiscovered_node_fill
+        self.completed_node_color=completed_node_color
+        self.undiscovered_edge_color=undiscovered_edge_color
+        self.discovered_edge_color=discovered_edge_color
+
+        self.add_node_colors()
+        self.add_edge_color()
+
+
+    def add_coords(self, coords):
+
+        node_data = []
+
+        if coords == None:
+            nodes_count = len(self.graph.get_graph_data())
+            while nodes_count:
+                node_data.append([(random.randint(self.node_radius,self.display_size[0]-self.node_radius), random.randint(self.node_radius,self.display_size[1]-self.node_radius))])
+                nodes_count -= 1
+        else:
+            tmp = sorted(coords.items())
+            for _, coord in tmp:
+                node_data.append([coord])
+            
+        return node_data
+    
+
+    def add_node_colors(self):
+        for node in range(0, len(self.nodes_data)):
+            self.nodes_data[node].extend([self.undiscovered_node_color, self.undiscovered_node_fill])
+
+    
+    def add_edge_color(self):
+        for edge in self.edges_data.keys():
+            self.edges_data[edge].append(self.undiscovered_edge_color)
+
+
+    def circle_fill(self, xy, line_color, fill_color, radius, thickness, num):
+        pygame.draw.circle(self.screen, line_color, xy, radius)
+        pygame.draw.circle(self.screen, fill_color, xy, radius - thickness)
+        self.screen.blit(num, num.get_rect(center = xy)) 
+
+    
+    def draw_graph(self):
+        font = pygame.font.SysFont(None, 25)
+
+        self.screen.fill((0, 0, 0,))
+
+        for e in self.edges_data.values(): # draw edges
+            (n1, n2), color = e
+            pygame.draw.line(self.screen, color, self.nodes_data[n1][0], self.nodes_data[n2][0], 2)
+
+            w = 0
+
+            for cost in self.graph.get_graph_data()[n1][0]:
+                if n2 in cost:
+                    w = cost[0]
+                
+            cost = font.render(f"{w}", True, self.labels_color)
+            self.screen.blit(cost, cost.get_rect(center = self.get_line_center(n1,n2)))
+
+            for n, (xy, lcolor, fcolor) in enumerate(self.nodes_data): # draw nodes
+                text = font.render(f"{n}", True, self.labels_color)
+                self.circle_fill(xy, lcolor, fcolor, self.node_radius, 2, text)
+
+
+    def get_line_center(self, n1 : int, n2 : int) -> float:
+        return ((self.nodes_data[n1][0][0]+self.nodes_data[n2][0][0])/2,(self.nodes_data[n1][0][1]+self.nodes_data[n2][0][1])/2)
```

Comment:
This line looks complicated enough, I would prefer to separate it to some variables